src/main/java/org/carlspring/cloud/storage/s3fs/S3SeekableByteChannel.java

```diff
@@ -137,18 +149,21 @@ protected void sync()
     {
         try (InputStream stream = new BufferedInputStream(Files.newInputStream(tempFile)))
         {
-            ObjectMetadata metadata = new ObjectMetadata();
-            metadata.setContentLength(Files.size(tempFile));
+            PutObjectRequest.Builder builder = PutObjectRequest.builder();
+            long length = Files.size(tempFile);
+            builder.contentLength(length);
 
             if (path.getFileName() != null)
             {
-                metadata.setContentType(new Tika().detect(stream, path.getFileName().toString()));
+                builder.contentType(new Tika().detect(stream, path.getFileName().toString()));
             }
 
-            String bucket = path.getFileStore().name();
-            String key = path.getKey();
+            builder.bucket(path.getFileStore().name());
+            builder.key(path.getKey());
+
+            S3Client client = path.getFileSystem().getClient();
 
-            path.getFileSystem().getClient().putObject(bucket, key, stream, metadata);
+            client.putObject(builder.build(), RequestBody.fromInputStream(stream, length));
```

Comment:
Multipart upload may be required if > 5GB

Reply:
To be implemented in #95.